### ISSUE TITLE
TransferUtility does not update the TransferListener when the preferred network type is not available

### ISSUE THREAD

**Issue by GerardoRobledo** [aws-amplify/aws-sdk-android#958]
**Describe the bug**
When using the following method ` public TransferObserver upload(String bucket, String key, File file, ObjectMetadata metadata,
            CannedAccessControlList cannedAcl, TransferListener listener)`  in airplane mode, `TransferStatusUpdater` won't notify any listener because is not added by the time `TransferRecord.checkPreferredNetworkAvailability` has been called.

**To Reproduce**
A code sample or steps:
```
 val observer = transferUtility.upload(Constants.BUCKET_NAME, fileToUpload.name, fileToUpload,
                    ObjectMetadata(),null,transferListener)
```

**Which AWS service(s) are affected?**
S3

**Expected behavior**
Debugging `TransferUtility` class you can see the following.
```
    public TransferObserver upload(String bucket, String key, File file, ObjectMetadata metadata,
            CannedAccessControlList cannedAcl, TransferListener listener) {
        if (file == null || file.isDirectory() || !file.exists()) {
            throw new IllegalArgumentException("Invalid file: " + file);
        }
        int recordId;
        if (shouldUploadInMultipart(file)) {
            recordId = createMultipartUploadRecords(bucket, key, file, metadata, cannedAcl);
        } else {

            final Uri uri = dbUtil.insertSingleTransferRecord(TransferType.UPLOAD, bucket, key, file, metadata,
                    cannedAcl, transferUtilityOptions);
            recordId = Integer.parseInt(uri.getLastPathSegment());
        }
        //this line triggers TransferRecord
       submitTransferJob(TRANSFER_ADD, recordId); 
       // this initialize our TransferObserver.
        return new TransferObserver(recordId, dbUtil, bucket, key, file, listener); 
    }
```

If  we change to 
```
    public TransferObserver upload(String bucket, String key, File file, ObjectMetadata metadata,
            CannedAccessControlList cannedAcl, TransferListener listener) {
        if (file == null || file.isDirectory() || !file.exists()) {
            throw new IllegalArgumentException("Invalid file: " + file);
        }
        int recordId;
        if (shouldUploadInMultipart(file)) {
            recordId = createMultipartUploadRecords(bucket, key, file, metadata, cannedAcl);
        } else {

            final Uri uri = dbUtil.insertSingleTransferRecord(TransferType.UPLOAD, bucket, key, file, metadata,
                    cannedAcl, transferUtilityOptions);
            recordId = Integer.parseInt(uri.getLastPathSegment());
        }
        TransferObserver transferObserver = new TransferObserver(recordId, dbUtil, bucket, key, file, listener)
        submitTransferJob(TRANSFER_ADD, recordId); 
       
        return transferObserver; 
    }
```

 then the listener would be called.

**Screenshots**
<img width="1680" alt="Screen Shot 2019-05-14 at 7 06 18 PM" src="https://user-images.githubusercontent.com/5215641/57740195-2b8a0a80-767c-11e9-9bbd-9c068fd8dd1d.png">
<img width="1792" alt="Screen Shot 2019-05-14 at 7 07 35 PM" src="https://user-images.githubusercontent.com/5215641/57740194-2b8a0a80-767c-11e9-9260-bc78437c6401.png">

<img width="1680" alt="Screen Shot 2019-05-14 at 7 08 48 PM" src="https://user-images.githubusercontent.com/5215641/57740193-2b8a0a80-767c-11e9-81e2-df7abf65cd3e.png">

**Comment by kvasukib** [comment]
@GerardoRobledo Thank you for the detailed analysis. I have submitted PR #961 to fix the issue. I will update this thread when the fix is released.

**Comment by kvasukib** [comment]
This has been fixed in `2.13.5` version of the SDK. closing the issue.
